src/js/sdk.tsx

```diff
@@ -54,40 +58,35 @@ export function init(passedOptions: ReactNativeOptions): void {
       ...DEFAULT_OPTIONS.transportOptions,
       ...(passedOptions.transportOptions ?? {}),
     },
-    integrations: getIntegrationsToSetup(passedOptions),
+    integrations: [],
     stackParser: stackParserFromStackParserOptions(passedOptions.stackParser || defaultStackParser)
   };
 
-  function addIntegration(integration: Integration): void {
-    if (options.integrations.filter((i) => i.name == integration.name).length === 0) {
-      options.integrations.push(integration);
-    }
-  }
```

Comment:
Adding this prevented adding duplicated integrations, I suspect this would introduce the bug again.
Here is the PR that added it, https://github.com/getsentry/sentry-react-native/pull/2319
Maybe try to reproduce the bug and see if still happens after this fix.

Reply:
Yes, I've seen it, but I think this function was only necessary because the `getIntegrationsToSetup` was called with wrong data too early.

There is this test for the bug and it passes.
https://github.com/getsentry/sentry-react-native/blob/15c1141e7e7fdce2874f0f9df6c5c0e7d4e67d32/test/sdk.test.ts#L112

Reply:
All good then, nice finding.